alea/src/testing.cpp

```diff
@@ -0,0 +1,53 @@
+#include <alps/alea/testing.hpp>
+
+namespace alps { namespace alea {
+
+template <typename T>
+t2_result t2_test(const column<T> &diff,
+                  const column<typename make_real<T>::type> &var,
+                  double nmeas, size_t pools, double atol)
+{
+//     std::cerr << "\nDIFF= " << diff.transpose()
+//               << "\nVAR=  " << var.transpose()
+//               << "\nNMEAS=" << nmeas << "\n";
+
+    if (diff.rows() != var.rows())
+        throw std::invalid_argument("Size mismatch between diff and var");
+    if (pools != 1 && pools != 2)
+        throw std::invalid_argument("Pools must be 1 or 2");
+    if (nmeas <= 0)
+        throw std::invalid_argument("Must have at least 1 measurement");
+    if ((var.array() < -atol).any())
+        throw std::invalid_argument("Variances must be positive");
```

Comment:
> According to your if condition, the variances must not be too negative. Therefore, the error message is a bit misleading. And, btw, why not just compare to 0?

Because these variances may come from a diagonalization step, which means that they might be zero within a numerical error. That is what is written here.